component/loki/source/kubernetes/kubetail/target.go

```diff
@@ -0,0 +1,236 @@
+package kubetail
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cespare/xxhash/v2"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Internal labels which indicate what container to tail logs from.
+const (
+	LabelPodNamespace     = "__pod_namespace__"
+	LabelPodName          = "__pod_name__"
+	LabelPodContainerName = "__pod_container_name__"
+	LabelPodUID           = "__pod_uid__"
```

Comment:
I might be missing something, but are these labels coming from somewhere, or a new fallback convention we'd like to enforce?

Reply:
Yeah, I'd like it to be the preferred convention since the `__meta_kubernetes` labels could be considered optional. We'll fall back to the meta labels if these don't exist though.